packages/wrangler/src/dev.tsx

```diff
@@ -506,9 +507,13 @@ export async function startDev(args: ArgumentsCamelCase<DevArgs>) {
 			watcher,
 			stop: async () => {
 				devReactElement.unmount();
-				await devReactElement.waitUntilExit();
 				await watcher?.close();
-				process.exit(0);
+			},
+			fetch: async () => {
+				const port = args.port || config.dev.port || (await getLocalPort());
```

Comment:
getLocalPort might not return the same value as was used by the server
I think you'll have to somehow get the value out

Reply:
ugh

Reply:
add a todo comment here, this will be critical to fix